src/confcom/azext_confcom/template_util.py

```diff
@@ -55,13 +55,17 @@ def case_insensitive_dict_get(dictionary, search_key) -> Any:
             return dictionary[key]
     return None
 
+
 def deep_dict_update(source: dict, destination: dict):
     """
     https://stackoverflow.com/questions/20656135/python-deep-merge-dictionary-data
     """
     for key, value in source.items():
         if isinstance(value, dict):
             node = destination.setdefault(key, {})
+            if node is None:
+                destination[key] = {}
+                node = destination
```

Comment:
In L68, should node be the new empty dict?

Reply:
it should, I think I was hitting a case where the value in `destination` was `None`. I'll look a little more closely at the issue. I don't think this is the best place for that error check